UPGRADE.md

```diff
@@ -0,0 +1,200 @@
+
+```typescript
+// Get a list of Calendars
+const calendars = nylas.calendars.list({
+  identifier: 'Grant_ID',
+});
+
+// Option 1 - Use the next() function to get the next page
+const nextPage = await calendars.next();
+
+// Option 2 - Use a for await...of loop to auto-paginate until the end
+for await (const item of calendars) {
+  // do something with each item
+}
+```
+
+### Error Objects
+
+Like the response objects, Nylas v3 now has standard error objects for all requests (excluding OAuth endpoints). There are two superclass error classes, `AbstractNylasApiError`, used for errors returned by the API, and `AbstractNylasSdkError`, used for errors returned by the SDK.
+
+The `AbstractNylasApiError` includes two subclasses: `NylasOAuthError`, used for API errors that are returned from the OAuth endpoints, and `NylasApiError`, used for any other Nylas API errors.
+
+The SDK extracts the error details from the response and stores them in the error object, along with the request ID and the HTTP status code.
+
+`AbstractNylasSdkError` is used for errors returned by the SDK. Right now there's only one type of error we return, and that's a `NylasSdkTimeoutError` which is thrown when a request times out.
+
+## Authentication
+
+The SDK's authentication methods reflect [the methods available in the new Nylas API v3](https://developer.nylas.com/docs/developer-guide/v3-authentication/). While you can only create and manage your application's connectors (formerly called integrations) in the dashboard, you can manage almost everything else directly from the SDK. This includes managing grants, redirect URIs, OAuth tokens, and authenticating your users.
+
+There are two main methods to focus on when authenticating users to your application. The first is the `Auth#urlForOAuth2` method, which returns the URL that you should redirect your users to in order to authenticate them using Nylas' OAuth 2.0 implementation.
+
+The second is the `Auth#exchangeCodeForToken` method. Use this method to exchange the code Nylas returned from the authentication redirect for an access token from the OAuth provider. Nylas's response to this request includes both the access token, and information about the grant that was created.  You don't _need_ to use the `grant_id` to make requests. Instead, you can use the authenticated email address directly as the identifier for the account. If you prefer to use the `grant_id`, you can extract it from the `CodeExchangeResponse` object and use that instead.
+
+The following code shows how to authenticate a user into a Nylas application:
+
+```typescript
+import Nylas from "nylas";
+
+const nylas = new Nylas({
+  api_key: "NYLAS_API_KEY",
+});
+
+// Build the URL for authentication
+const authURL = nylas.auth.urlForOAuth2({
+  clientId: "CLIENT_ID",
+  redirectUri: "REDIRECT_URI",
```

Comment:
Same question as in the other repo. I think this is the Nylas redirect_uri, and the other is the callback.

Reply:
As per our conversation, we're holding off on mentioning the readme in the SDKs until the API team deploys the changes to staging and docs are updated. Once the contract is cemented, we'll update the SDK and these readme docs, thanks!